config/opal_check_cuda.m4

```diff
@@ -50,9 +50,18 @@ AC_ARG_WITH([cuda],
             [AS_HELP_STRING([--with-cuda(=DIR)],
             [Build cuda support, optionally adding DIR/include])])
 AC_MSG_CHECKING([if --with-cuda is set])
+
+# Search for libcuda.so in $with_cuda if the user didn't pass --with-cuda-libdir
+# Otherwise check for cuda in the default path, /usr/local/cuda. If the default
+# path doesn't exist, set with_cuda_libdir to empty.
 AC_ARG_WITH([cuda-libdir],
             [AS_HELP_STRING([--with-cuda-libdir=DIR],
-                            [Search for CUDA libraries in DIR])])
+                            [Search for CUDA libraries in DIR])],
+            [],
+            [AS_IF([test -d "$with_cuda"],
+             [with_cuda_libdir=$(dirname $(find $with_cuda -name libcuda.so) 2> /dev/null)],
```

Comment:
Is there ever a case where there's more than one `libcuda.so` in that tree?

Is `libcuda.so` always in the same location in that tree?  If so, should you just look for it in `$with_cuda/FIXED_LOCATION/libcuda.so`?

Reply:
I don't know if it's always in the same location, for now it seems to be with_cuda/lib64/stubs, but why limit it to a fixed location if it ever changes.

Reply:
>Is there ever a case where there's more than one libcuda.so in that tree?
Not sure, I've never encountered this case, but... *shrug*